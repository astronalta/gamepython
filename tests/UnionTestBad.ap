

TestInterface < Interface {
}

TestUnion = String | Int
TestUnion2 = String | TestInterface


function(bool Bool) {

}

main() Int {

    int1 TestInterface
    test1 TestUnion = true
    test2 TestUnion = int1
    
    bool Bool = test1
    int2 TestInterface = test1

    function(test1)

    test1 is TestInterface
    test1 is TestUnion
    test1 is Any 

    ret 0
}
