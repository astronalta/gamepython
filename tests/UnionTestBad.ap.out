UnionTestBad.ap:7:23: Interface cannot be used as an alternate
UnionTestBad.ap:17:23: Expression does not conform to type 'TestUnion'
UnionTestBad.ap:18:23: Expression does not conform to type 'TestUnion'
UnionTestBad.ap:20:17: Expression does not conform to type 'Bool'
UnionTestBad.ap:21:26: Expression does not conform to type 'TestInterface'
UnionTestBad.ap:23:14: Argument does not conform to type 'Bool'
UnionTestBad.ap:25:14: Type in 'is' expression cannot be an interface
UnionTestBad.ap:26:14: Type in 'is' expression cannot be a union
UnionTestBad.ap:27:14: Type in 'is' expression cannot be 'Any'
UnionTestBad.ap:28:14: Type in 'is' expression cannot be parameterized
UnionTestBad.ap:31:23: Expression does not conform to type 'TestUnion'
